test/refactoring/AddCodableImplementation/Outputs/basic/codable.swift.expected

```diff
@@ -0,0 +1,35 @@
+
+struct User: Codable {
+  let firstName: String
+  let lastName: String?
+
+  enum CodingKeys: CodingKey {
+    case firstName
+    case lastName
+  }
+
+  init(from decoder: Decoder) throws {
+    let container: KeyedDecodingContainer<User.CodingKeys> = try decoder.container(keyedBy: User.CodingKeys.self)
+
+    self.firstName = try container.decode(String.self, forKey: User.CodingKeys.firstName)
+    self.lastName = try container.decodeIfPresent(String.self, forKey: User.CodingKeys.lastName)
+    return
```

Comment:
The `return` at the end is redundant of course. I'm not sure how to best handle that in the ASTPrinter. (we'd have to check if it's the last statement in the scope, I suppose). 

For now, I think it's fine to just keep it. It compiles, but looks a bit weird.

Reply:
Is it implicit? If it isn't, could be be? And if so, can we skip printing implicit expressions?

Reply:
I ended up adding a (very specific) `SkipImplicitReturn` option to `PrintOptions`. Skipping _all_ implicit `return` stmts made an existing test fail, so I wanted to minimize the potential breakage.

Reply:
Yeah I'm not sure skipping all implicit returns is correct, as implicit just means "created by the compiler", and we shouldn't skip an implicit return in e.g `if condition { return }`, or one with an expression e.g `return 0`. Would it be possible to narrow this behavior by having the AST printer only drop an implicit return stmt that is the last element of an AbstractFunctionDecl and has no expression?

Reply:
> having the AST printer only drop an implicit return stmt that is the last element of an AbstractFunctionDecl and has no expression?

Ah, good idea. I've done that and removed the option I added (so it will just always skip the printing of such `return`s).